scripts/build/templates/icon-object.js

```diff
@@ -1,10 +1,10 @@
 {
   title: '%s',
   slug: '%s',
-  svg: '%s',
-  get path() {
-    return this.svg.match(/<path\s+d="([^"]*)/)[1];
+  get svg() {
+    return '<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>%s</title><path d="' + this.path + '"/></svg>';
```

Comment:
Can we use `this.title` here instead of escaping the title and duplicating it?

Reply:
> Can we use `this.title` here instead of escaping the title and duplicating it? 

I've tested it and it is slower, so I prefer to maintain the string as it is even if we could decrease the size of files using `this.title`.

> You may want to use a template literal to make it look better.

Template literals are an ES6 feature right? It seems that [is not universally supported](https://caniuse.com/?search=string%20template%20literals), so I prefer to maintain the compatibility for all platforms and NodeJS versions. This would be a breaking change.

Reply:
>>  Can we use `this.title` here instead of escaping the title and duplicating it? 
> I've tested it and it is slower, so I prefer to maintain the string as it is even if we could decrease the size of files using `this.title`.

Just seeing now [@ericcornelissen comment in issue](https://github.com/simple-icons/simple-icons/issues/6481#issuecomment-917197995) regardling this. I want to share here the performance and file size tests that I've done because to take the current approach of duplicate the title:

```sh
$ time node -e "for (let i=0; i<500000; i++) { require('./icons/elsevier').svg }"

real	0m0,152s
user	0m0,149s
sys	0m0,015s
$ cat icons/elsevier.js | wc --bytes
54338
$ sed -i "s/<title>Elsevier<\/title>/<title>\'+this.title+\'<\/title>/g" icons/elsevier.js
$ time node -e "for (let i=0; i<500000; i++) { require('./icons/elsevier').svg }"

real	0m0,172s
user	0m0,170s
sys	0m0,013s
$ cat icons/elsevier.js | wc --bytes
54344
$ sed -i "s/<title>'+this.title+'<\/title>/<title>Elsevier<\/title>/g" icons/elsevier.js
```

I'm on Ubuntu 20.04. I suppose that it is replicable in other machines as well. If, even knowing this performance decreasement and that the file size would be greater for icons whose title has minus than 14 characters ("Elsevier" has 8 and the difference is +6 changing by `'+this.title+'`) do you want to go with that change, I'm open to it, but IMHO I think it is unnecessary in this case.

Reply:
I don't think saving a few bytes will really matter when we're already in the tens of thousands. I do see a slight performance hit (also on Ubuntu 20.04), so I'm okay with inlining `this.title`. I think we should wait for a third opinion though.

Reply:
> Template literals are an ES6 feature right? It seems that is not universally supported, so I prefer to maintain the compatibility for all platforms and NodeJS versions. This would be a breaking change.

We should indeed avoid using template literals until we're ready to introduce a breaking change that bumps up the minimum version of JS significantly. We _could_ use it to make the source look cleaner if we can configure, e.g., UglifyJS to convert template literals to string concatenation. But not sure if that's worth the effort.


> I want to share here the performance and file size tests that I've done before to take the current title duplication approach: [...]

Thanks for sharing @mondeja, that's very useful. I think I agree that embedding the title value directly is the better option here as it improves performance. Duplication is not a concern here as we generate the code and when considering all brands I'm 1) not even sure using `this.title` would decrease the package size and 2) of the opinion that the performance increase is more significant.

Reply:
I think we should save the escaped title to a variable though (in `iconToObject`) just to avoid escaping it twice.

Reply:
Actually, the title value in the `<title>` probably shouldn't be escaped

Reply:
Done in https://github.com/simple-icons/simple-icons/pull/6505/commits/4e375963525b9795b1f2b5f193f538c2429829bf :+1: